### PR TITLE
added testcase for vtgate vars

### PR DESCRIPTION
**Pull request by arindamnayak**
Signed-off-by: Arindam Nayak <arindam.nayak@outlook.com>

@deepthi , this testcase corresponds to this one - https://github.com/planetscale/vitess/blob/master/test/vtgatev3_test.py#L2020

### CODE REVIEW

File: go/test/endtoend/clustertest/vtgate_test.go [hunk 1]
```diff
@@ -42,6 +48,64 @@ func TestVtgateProcess(t *testing.T) {
 	}
 }
 
+func verfiyVtgateVariables(t *testing.T, url string) {
```
Comment (deepthi) [inline go/test/endtoend/clustertest/vtgate_test.go]: fix function name: verfiy -> verify

File: go/test/endtoend/clustertest/vtgate_test.go [hunk 2]
```diff
@@ -42,6 +48,64 @@ func TestVtgateProcess(t *testing.T) {
...
+	resp, _ := http.Get(url)
+	if resp != nil && resp.StatusCode == 200 {
```
Comment (deepthi) [inline go/test/endtoend/clustertest/vtgate_test.go:53]: test should fail if resp is nil? or statusCode != 200?

File: go/test/endtoend/clustertest/vtgate_test.go [hunk 3]
```diff
@@ -42,6 +48,64 @@ func TestVtgateProcess(t *testing.T) {
...
+			t.Error("There should not be any Parsing in VschemaCount")
+		}
+
+		if resultMap["HealthcheckConnections"] == nil {
+			t.Error("HealthcheckConnections count should be present in variables")
+		}
+
+		healthCheckConnection := getMapFromJSON(resultMap, "HealthcheckConnections")
+		if len(healthCheckConnection) <= 0 {
+			t.Error("Atleast one healthy tablets need to be present")
```
Comment (deepthi) [inline go/test/endtoend/clustertest/vtgate_test.go]: tablets -> tablet, need -> needs

File: go/test/endtoend/clustertest/vtgate_test.go [hunk 4]
```diff
@@ -42,6 +48,64 @@ func TestVtgateProcess(t *testing.T) {
...
+		}
+		if !isMasterTabletPresent(healthCheckConnection) {
+			t.Error("Atleast one master tablet need to be present")
```
Comment (deepthi) [inline go/test/endtoend/clustertest/vtgate_test.go]: need -> needs

File: go/test/endtoend/clustertest/vtgate_test.go [hunk 5]
```diff
@@ -42,6 +48,64 @@ func TestVtgateProcess(t *testing.T) {
...
+		respByte, _ := ioutil.ReadAll(resp.Body)
+		err := json.Unmarshal(respByte, &resultMap)
+		if err != nil {
+			t.Fatal(err)
+		}
+		if resultMap["VtgateVSchemaCounts"] == nil {
+			t.Error("Vschema count should be present in variables")
+		}
+		vschemaCountMap := getMapFromJSON(resultMap, "VtgateVSchemaCounts")
+		if _, present := vschemaCountMap["Reload"]; !present {
```
Comment (deepthi) [inline go/test/endtoend/clustertest/vtgate_test.go:64]: The original test checks for Reload > 0. Is that possible here?

**Comment by arindamnayak** [comment]
@deepthi , all changes implemented in last commit.

**Review by deepthi — APPROVED** [review]
Looks good.

**Comment by arindamnayak** [comment]
Once https://github.com/vitessio/vitess/pull/5346 is merged, I will merge this one to appropriate branch.

**Comment by arindamnayak** [comment]
I have merged this change to tal_test_master manually. Hence closing this PR.

**Pull request closed without merging**